hnswalg.cpp

```diff
@@ -233,11 +250,56 @@ dist_t fstdistfunc_scalar(const coord_t *x, const coord_t *y, size_t n)
 
 }
 
+dist_t dist_manhattan_scalar(const coord_t *x, const coord_t *y, size_t n)
+{
+    dist_t 	distance = 0.0;
+
+    #pragma clang loop vectorize(enable)
+    for (size_t i = 0; i < n; i++)
+    {
+        dist_t diff = x[i] - y[i];
```

Comment:
Unless I'm mistaken, the L1 norm is with the absolute value of the difference. Does the `dist_t` type automatically do that?

Reply:
when we calculate diff*diff sign of diff doesn't matter

Reply:
Sorry, I looked in wrong place

Reply:
> the L1 norm is with the absolute value of the difference

Good catch, thanks! I've fixed it

Reply:
Sure, but you missed `dist_manhattan_avx2` as well.

Reply:
No I didn't:
`__m256 vecAbs = _mm256_andnot_ps(vecMask, vecSub);`